test/proxy-bcrypt.test.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const { createUnzip } = require('zlib');
+
+const tar = require('tar-fs');
+const Agent = require('https-proxy-agent');
+const fetch = require('node-fetch');
+const rimraf = require('rimraf');
+
+const test = require('tape');
+
+// this is a derived from build.test.js and should be kept in sync with it
+// as much as possible.
+const { mockS3Http } = require('../lib/node-pre-gyp.js');
+const proxy = require('./proxy.util');
+
+const proxyPort = 8124;
+const proxyServer = `http://localhost:${proxyPort}`;
+// options for fetch
+const options = {};
+
+let initial_s3_host;
+let initial_mock_s3;
+
+// the temporary download directory and file
+const downloadDir = 'download';
+
+// https://stackoverflow.com/questions/38599457/how-to-write-a-custom-assertion-for-testing-node-or-javascript-with-tape-or-che
+test.Test.prototype.stringContains = function(actual, contents, message) {
+  this._assert(actual.indexOf(contents) > -1, {
+    message: message || 'should contain ' + contents,
+    operator: 'stringContains',
+    actual: actual,
+    expected: contents
+  });
+};
+
+//
+// skip tests that require a real S3 bucket when in a CI environment
+// and the AWS access key is not available.
+//
+const isCI = process.env.CI && process.env.CI.toLowerCase() === 'true'
+  && !process.env.AWS_ACCESS_KEY_ID;
+
+function ciSkip(...args) {
+  if (isCI) {
+    test.skip(...args);
+  } else {
+    test(...args);
+  }
+}
+ciSkip.skip = function(...args) {
+  test.skip(...args);
+};
+
+test('setup proxy server', (t) => {
+  delete process.env.http_proxy;
+  delete process.env.https_proxy;
+  delete process.env.HTTPS_PROXY;
+  delete process.env.all_proxy;
+  delete process.env.ALL_PROXY;
+  delete process.env.no_proxy;
+  delete process.env.NO_PROXY;
+  process.env.NOCK_OFF = true;
+
+  initial_mock_s3 = process.env.node_pre_gyp_mock_s3;
+  delete process.env.node_pre_gyp_mock_s3;
+  mockS3Http('off');
+
+  proxy.startServer({ port: proxyPort });
+  process.env.https_proxy = process.env.http_proxy = proxyServer;
+
+  options.agent = new Agent(proxyServer);
+
+  process.env.NOCK_OFF = true;
+
+  // make sure the download directory deleted then create an empty one
+  rimraf(downloadDir, () => {
+    fs.mkdir('download', (e) => {
+      if (e && e.code !== 'EEXIST') {
+        t.error(e);
+        return;
+      }
+      t.end();
+    });
+  });
+
+
+});
+
+test('verify node fetch with a proxy successfully downloads bcrypt pre-built', (t) => {
+  // "{module_name}-v{version}-napi-v{napi_build_version}-{platform}-{arch}-{libc}.tar.gz"
+  const url = 'https://github.com/kelektiv/node.bcrypt.js/releases/download/v5.0.1/bcrypt_lib-v5.0.1-napi-v3-linux-x64-glibc.tar.gz';
```

Comment:
/cc @kelektiv - so that you are aware that we're building logic into our tests that reference your downloads. Do you have any concern with this? (e.g. hopefully you don't plan to remove files like this in the future and rather they will stay around indefinitely?)

Reply:
worst cases:
1. we can update the tests if they remove that particular release
2. we can add logic to find a recent release and fetch that; i wanted to defer that so we could get a fix out but it could be follow-on work if need be